### PR TITLE
Add plotting extension

### PR DESCRIPTION
**Pull request by laurent-dinh**
Plotting extension : create png files containing plots for chosen monitoring channels.

### CODE REVIEW

File: pylearn2/train_extensions/plots.py [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+        Parameters
+        ----------
+        model : pylearn2.models.Model
+            The model trained
+        dataset : pylearn2.datasets.Dataset
+            The dataset on which the model is trained
+        algorithm : pylearn2.training_algorithms.TrainingAlgorithm
+            The algorithm the model is trained with
+
+        """
```
Comment (lamblin) [inline pylearn2/train_extensions/plots.py]: Unfortunately, it is not possible.

**Comment by lamblin** [comment]
Regarding the back-end selection, instead of always using Agg, you can check whether the `DISPLAY` environment variable is set: http://stackoverflow.com/questions/8257385/automatic-detection-of-display-availability-with-matplotlib
If it is not set, then use Agg, otherwise keep the default.

**Comment by laurent-dinh** [comment]
Updated.

File: pylearn2/train_extensions/plots.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+        raise NotImplementedError(str(type(self))+" does not implement setup.")
+
+    def plot(self):
+        """
+        Plot what it should.
```
Comment (lamblin) [inline pylearn2/train_extensions/plots.py]: ಠ_ಠ

**Comment by lamblin** [comment]
Thanks, just let me know when you have updated the documentation of `plot`, and manually checked that we have the desired behaviour:
- on a local machine with X, the default back-end is used
- on a remote connection with X (`ssh -X`), the default back-end is used
- on a headleass remote connection, the "Agg" back-end is used.

**Comment by lamblin** [comment]
Thanks!

**Pull request merged**